march_hardware/src/Joint.cpp

```diff
@@ -50,34 +55,62 @@ void Joint::actuateRad(double targetPositionRad)
   this->iMotionCube.actuateRad(targetPositionRad);
 }
 
-double Joint::getAngleRadAbsolute()
+void Joint::readEncoders(const ros::Duration& elapsed_time)
 {
-  if (!hasIMotionCube())
+  const double new_incremental_position = this->iMotionCube.getAngleRadIncremental();
+  const double new_absolute_position = this->iMotionCube.getAngleRadAbsolute();
+
+  // Get velocity from encoder position
+  const double incremental_velocity = (new_incremental_position - this->incremental_position_) / elapsed_time.toSec();
+  const double absolute_velocity = (new_absolute_position - this->absolute_position_) / elapsed_time.toSec();
+
+  if (std::abs(incremental_velocity - absolute_velocity) * elapsed_time.toSec() >
```

Comment:
This if statement is quit unreadable. I think you should either add a comment on what it is doing or break it up so that the resulting if statement looks like `if position_difference > threshold`. Furthermore Some parts don't need to be calculated over and over again. The second part of the comparison is constant.

Reply:
I added a comment :+1: